src/helpers/filesystem.ts

```diff
@@ -6,6 +6,50 @@ import { yellow, green } from "./write-help";
 
 export const templatesSourcePath = path.join(__dirname, "../../templates/");
 
+/**
+ * Validate `destination` path, throw error if not valid (e.g not an empty folder)
+ * @param {string} destination  destination path
+ * @param {boolean} overwrite   if true, don't throw error if path is a non-empty valid folder
+ */
+export async function ensureValidDestination(
+  destination: string,
+  overwrite: boolean
+): Promise<void> {
+  const invalidDestinationError: Error = new Error(`Invalid destination folder => ${destination}
+Please provide either an empty folder or a non existing path as <destination>
+`);
+
+  try {
+    fs.lstatSync(destination);
+
+    try {
+      // edge case: destination might be a broken link
+      fs.realpathSync(destination);
+    } catch (error) {
+      if (error.code === "ENOENT") throw invalidDestinationError;
+    }
+  } catch (error) {
+    if (error.code === "ENOENT") return;
+    // not-existing path
+    else throw error;
+  }
+
+  if (fs.statSync(destination).isDirectory()) {
+    if (fs.readdirSync(destination).length === 0) return; // empty folder
+    if (overwrite) {
+      console.warn(
+        yellow(`Explicit OVERWRITE option selected:
+Some files under "${fs.realpathSync(destination)}" might be overwritten!
+`)
+      );
+      return;
+    }
+  }
+
+  // either not a folder or a not empty one
```

Comment:
nit: Add a subject for clarification.

```suggestion
  // The `destination` is either not a folder or a not empty one.
```